plugins/basemap-gaode-by-gmoogway.user.js

```diff
@@ -0,0 +1,351 @@
+            var lat_deg_arclen = (Math.PI / 180) * (GCJ_A * (1 - GCJ_EE)) * Math.pow(magic, 1.5)
+            var lon_deg_arclen = (Math.PI / 180) * (GCJ_A * Math.cos(radLat) / Math.sqrt(magic))
+            if (resetting){
+                return {
+                    lat: wgs.lat - (dLat_m / lat_deg_arclen),
+                    lng: wgs.lng - (dLon_m / lon_deg_arclen),
+                }
+            }else{
+                return {
+                    lat: wgs.lat + (dLat_m / lat_deg_arclen),
+                    lng: wgs.lng + (dLon_m / lon_deg_arclen),
+                }
+            }
+
+        }
+
+        return {
+            gcjObfs: gcjObfs,
+            isInGoogle: isInGoogle,
+        }
+    })();
+
+    /////////// end WGS84 to GCJ-02 obfuscator /////////
+
+    window.plugin.mapTileGaode.addLayer = function() {
+
+        var types = {
+            'Normal': ['路网', 3, 20, 'appmaptile?x={x}&y={y}&z={z}&lang=zh_cn&size=1&scl=1&style=7', 'GD-Normal'],
+            'Satellite': ['影像', 3, 18, 'appmaptile?x={x}&y={y}&z={z}&lang=zh_cn&size=1&scl=1&style=6', 'GD-Satellite'],
+        };
+
+        var baseUrl = 'https://wprd0{s}.is.autonavi.com/';
+
+        for (var layer in types) {
+            var info = types[layer];
+
+            var name = info[0];
+            var minZoom = info[1];
+            var maxZoom = info[2];
+            var address = info[3];
+
+            var mapLayer = new L.TileLayer(baseUrl + address, {
+                attribution: 'map tiles by autonavi.com, plugin written by GMOogway.',
+                subdomains: '1234',
+                layer: layer,
+                type: '',
+                minZoom: minZoom,
+                maxNativeZoom: maxZoom,
+                maxZoom: 20
+            });
+
+            layerChooser.addBaseLayer(mapLayer, '高德 ' + name);
+        }
+
+    };
+
+    window.plugin.mapTileGaode.drawbutton = function() {
+
+    }
+
+    var setup = function() {
+
+        window.plugin.mapTileGaode.addLayer();
+
+        window.plugin.mapTileGaode.drawbutton();
+
+        if (!localStorage['iitc-base-map-gaode-alert'] || localStorage['iitc-base-map-gaode-alert'] != 1 ){
+            alert('After the first use this plugin, please refresh it manually due to the need to correct the offset. At present, the coordinates of portal, link, field, PlayerTracker, BookMarks have been fixed. Because iitc-mobile on iPhone can\'t select the layer in the new version of IITC, please remove the following comments before adding to user script. If you want to use other maps, disable this plugin. Map tiles by autonavi.com, plugin written by (TG)GMOogway.');
+            localStorage['iitc-base-map-gaode-alert'] = 1;
+        }
+
+        // for iitc-mobile on ios
+        //localStorage['iitc-base-map'] = '高德 路网';
+
+        var newpos;
+
+        var PluginBookmarksExist = false;
+        var PluginPlayerTrackerExist = false;
+        if (window.bootPlugins) {
```

Comment:
This is wrong approach, because this code may be executed before that plugins loaded.

Reply:
fixed